go/vt/worker/split_diff.go

```diff
@@ -239,14 +239,29 @@ func (sdw *SplitDiffWorker) findTargets(ctx context.Context) error {
 	// find an appropriate tablet in the source shard
 	for _, ss := range sdw.shardInfo.SourceShards {
 		if ss.Uid == sdw.sourceUID {
-			sdw.sourceAlias, err = FindWorkerTablet(ctx, sdw.wr, sdw.cleaner, nil /* tsc */, sdw.cell, sdw.keyspace, ss.Shard, sdw.minHealthyRdonlyTablets, topodatapb.TabletType_RDONLY)
-			if err != nil {
-				return fmt.Errorf("FindWorkerTablet() failed for %v/%v/%v: %v", sdw.cell, sdw.keyspace, ss.Shard, err)
+			// During an horizontal shard split, multiple workers will race to get
+			// a RDONLY tablet in the source shard. When this happen, concurrent calls
+			// to FindWorkerTablet could attempt to set to DRAIN state the same tablet. Only
+			// one of these calls to FindWorkerTablet will succeed and the rest will fail.
+			// The following, makes sures we keep trying to find a worker tablet when this error occur.
+			shortCtx, cancel := context.WithTimeout(ctx, *remoteActionsTimeout)
+			for {
```

Comment:
I'm not sure this loop will help.

The context gets passed in, and the downstream calls are themselves looping waiting for context to expire. I see one difference in `waitForHealthyTablets` where a `waitForHealthyTabletsTimeout` gets added, but its default value is same as `remoteActionsTimeout`.

Can you check the error that caused the race? Then we can identify the code path, and ideally fix it at the point where the error origintated.

Reply:
Hey Sugu - I've validated in my local environment that this fixes the problem.

The loop by itself does not help. The fix is the loop + the change where now when you try to call `drain` on a tablet that is already drained, you will get an error. 

Before this change, the race allow two vtworkers to use the same tablet and if I recall correctly one of the actual errors you get is when trying to synchronizeReplication. Both workers are trying to StopBlp and they get errors.